core/src/main/java/de/danoeh/antennapod/core/service/FeedUpdateJobService.java

```diff
@@ -17,8 +17,13 @@
     public boolean onStartJob(JobParameters params) {
         Log.d(TAG, "Job started");
         ClientConfig.initialize(getApplicationContext());
-        FeedUpdateUtils.startAutoUpdate(getApplicationContext());
-        UserPreferences.restartUpdateAlarm(false);
+
```

Comment:
The codes here (spawning new thread, using synchronous version of startAutoUpdate(), etc.) are a tad tricky. 

A way to reduce the complexity could be changing `FeedUpdateUtils.startAutoUpdate()` to accept a `Runnable onCompleteCallback` parameter, so that

1. At `FeedUpdateJobService` level, it simply calls: 
```java
        FeedUpdateUtils.startAutoUpdate(getApplicationContext(), () -> {
           UserPreferences.restartUpdateAlarm(false);
           jobFinished(params, false); // needsReschedule = false
        });
```

2. In the underlying DBTasks.refreshAllFeeds(context, feeds), accept the new parameter `Runnable onCompleteCallback`, and invokes the callback upon completion.
    - `DBTasks` and `FeedUpdateUtils` thus have no need to expose synchronous 
 version of feed updates.